### ISSUE TITLE
Translate STL to (0,0,0)

### ISSUE THREAD

**Issue by nwagenbrenner** [firelab/windninja#120]
ninjaElevationToStl should translate the min bounds to (0,0,0) at creation time. Then we just need to translate (the 2 AsciiGrids created from the raw file) back to DEM coordinates after we sample the simpleFoam output.

**Comment by ksshannon** [comment]
So we just drop the offsets?  This shouldn't be too hard.

**Comment by nwagenbrenner** [comment]
We just need xmin, ymin, zmin to be 0, 0, 0 in the STL. Then after we read in the raw output (from simpleFoam), we just need to shift the speed and direction ascii grids back to the DEM coordinates (the real coordinates of the coarsened STL surface) so we can write our usual outputs. So yeah, should just be translating, no changing of cell sizes or anything.

**Comment by ksshannon** [comment]
I attempted to do this, and it causes refineMesh to hang.  Does this need to be in ninjafoam-upgrade?

**Comment by nwagenbrenner** [comment]
Yeah, sorry, this should be in ninjafoam-upgrade. It probably won't just work in that branch yet either though, I'm working on the meshing stuff right now... I have some stuff hard coded in there for Big Butte, but am updating it now.

**Comment by ksshannon** [comment]
Also, along those lines, I probably can't merge recent trunk changes into that branch.

**Comment by nwagenbrenner** [comment]
What do you mean? I was sort of hoping to have all the new meshing stuff sorted out and working next week so we could merge it back in to the master branch...

**Comment by ksshannon** [comment]
I can merge master into ninjafoam-upgrade then?  The stl coarsening fix was in master.

**Comment by nwagenbrenner** [comment]
Yeah, I think so. I saw the coarsening fix, thanks. Try to merge and if there are problems, let me know and I can deal with it.

**Comment by ksshannon** [comment]
So changing the origin to (0,0,0) causes refineMesh to hang indefinitely.  Merging master -> ninjafoam-upgrade has a snappyHexMesh Error().  Now I'm fighting with git to get it back to the original state.  I'll post a patch instead of committing, if I can get git to cooperate.

**Comment by nwagenbrenner** [comment]
I would expect an error in SHM since I have hard-coded stuff in there for BSB right now. The merge might actually be okay... But,  yeah, just do whichever you think is easiest/best.

**Comment by ksshannon** [comment]
Merging master into your branch goes okay, but errors on SHM even when using big_butte.tif.  If it's okay, I'll merge master, make the (0,0,0) changes and check it in.  We may have to fix it Monday though.  That okay?

**Comment by nwagenbrenner** [comment]
Yeah, I think that's okay. I'm not surprised there are still errors, just because of the way have things hard-coded in there. It should be fine to just merge your changes. Thanks.

**Comment by ksshannon** [comment]
Okay, sounds good.

**Comment by ksshannon** [comment]
Checked in to ninjafoam-upgrade.  Not totally sure I did that totally right, so test and post the results Monday or so...

**Comment by nwagenbrenner** [comment]
I think this isn't quite right, at least in the y-direction. Output stats from surface check on the stl:

Statistics:
Triangles    : 5292
Vertices     : 2750
Bounding Box : (15.4618 -8115.46 1678) (7365.46 -15.4618 2444)

**Comment by ksshannon** [comment]
Use AsciiGrid, see #119

**Comment by nwagenbrenner** [comment]
We should try to get this done for 3.2.0. It's possible that precision issues have been causing a number of previously reported problems (including mesh manipulation near the surface).
